### ISSUE TITLE
Push cancelled enrollments before

### ISSUE THREAD

**Issue by idelcano** [EyeSeeTea/dhis2-android-trackercapture#117]

**Comment by ifoche** [comment]
- We've tested postEnrollmentsBatch() and looks like it's failing with an exception in retrofit when mapping the ImportSummary
```
07-07 01:16:53.751 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: ---> HTTP POST https://play.dhis2.org/android-previous1/api/enrollments/
07-07 01:16:53.752 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Content-Type: application/json; charset=UTF-8
07-07 01:16:53.752 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Content-Length: 396
07-07 01:16:53.752 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: {"attributes":[{"value":"ina007","attribute":"w75KJ2mc4zz"},{"value":"Bond","attribute":"zDhUuAYrxNC"},{"value":"Male","attribute":"cejWyOfXge6"},{"value":"1472514437","attribute":"xs8A6tQJY0s"}],"enrollment":"bcN5OJSyOMa","enrollmentDate":"2017-07-07T01:00:00.000+02:00","followup":false,"orgUnit":"DiszpKrYNg8","program":"ur1Edk5Oe2n","status":"CANCELLED","trackedEntityInstance":"AHXw0cLdpjI"}
07-07 01:16:53.752 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: ---> END HTTP (396-byte body)
07-07 01:16:53.753 30448-31087/org.hisp.dhis.android.trackercapture D/RepoManager: Request{method=POST, url=https://play.dhis2.org/android-previous1/api/enrollments/, tag=Request{method=POST, url=https://play.dhis2.org/android-previous1/api/enrollments/, tag=null}}
07-07 01:16:53.953 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: <--- HTTP 409 https://play.dhis2.org/android-previous1/api/enrollments/ (200ms)
07-07 01:16:53.953 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Server: nginx/1.4.6 (Ubuntu)
07-07 01:16:53.954 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Date: Thu, 06 Jul 2017 23:18:16 GMT
07-07 01:16:53.954 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Content-Type: application/json;charset=UTF-8
07-07 01:16:53.954 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Content-Length: 897
07-07 01:16:53.954 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Connection: keep-alive
07-07 01:16:53.954 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: X-XSS-Protection: 1; mode=block
07-07 01:16:53.955 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: X-Frame-Options: SAMEORIGIN
07-07 01:16:53.955 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: X-Content-Type-Options: nosniff
07-07 01:16:53.955 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Set-Cookie: JSESSIONID=F005AA0C6AF967B88CC6942676C4B1CA; Path=/android-previous1; HttpOnly
07-07 01:16:53.955 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: Cache-Control: no-cache, private
07-07 01:16:53.960 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: OkHttp-Sent-Millis: 1499383013788
07-07 01:16:53.960 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: OkHttp-Received-Millis: 1499383013953
07-07 01:16:53.961 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: {"httpStatus":"Conflict","httpStatusCode":409,"status":"WARNING","message":"One more conflicts encountered, please check import summary.","response":{"responseType":"ImportSummaries","status":"SUCCESS","imported":0,"updated":0,"deleted":0,"ignored":1,"importOptions":{"idSchemes":{},"dryRun":false,"async":false,"importStrategy":"CREATE","mergeMode":"REPLACE","skipExistingCheck":false,"sharing":false,"skipNotifications":false,"datasetAllowsPeriods":false,"strictPeriods":false,"strictCategoryOptionCombos":false,"strictAttributeOptionCombos":false,"strictOrganisationUnits":false,"requireCategoryOptionCombo":false,"requireAttributeOptionCombo":false},"importSummaries":[{"responseType":"ImportSummary","status":"ERROR","description":"TrackedEntityInstance AHXw0cLdpjI already have an active enrollment in program ur1Edk5Oe2n","importCount":{"imported":0,"updated":0,"ignored":1,"deleted":0}}]}}
07-07 01:16:53.961 30448-31087/org.hisp.dhis.android.trackercapture D/Retrofit: <--- END HTTP (897-byte body)
07-07 01:16:53.961 30448-31087/org.hisp.dhis.android.trackercapture W/System.err: retrofit.RetrofitError: 409 
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at java.lang.reflect.Proxy.invoke(Proxy.java:813)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at $Proxy2.postEnrollment(Unknown Source)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.postEnrollment(TrackerDataSender.java:427)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.sendEnrollmentChanges(TrackerDataSender.java:411)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.sendEventChanges(TrackerDataSender.java:216)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.sendEventChanges(TrackerDataSender.java:193)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.postEventBatch(TrackerDataSender.java:172)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.sendEventBatch(TrackerDataSender.java:130)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerDataSender.sendEventChanges(TrackerDataSender.java:114)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.tracker.TrackerController.sendLocalData(TrackerController.java:550)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.DhisController.sendData(DhisController.java:137)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.DhisController.synchronize(DhisController.java:118)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.controllers.DhisService$8.execute(DhisService.java:176)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.job.NetworkJob.inBackground(NetworkJob.java:51)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.job.NetworkJob.inBackground(NetworkJob.java:38)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.job.Job.doInBackground(Job.java:60)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at org.hisp.dhis.android.sdk.job.Job.doInBackground(Job.java:36)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at android.os.AsyncTask$2.call(AsyncTask.java:305)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
07-07 01:16:53.962 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
07-07 01:16:53.963 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
07-07 01:16:53.963 30448-31087/org.hisp.dhis.android.trackercapture W/System.err:     at java.lang.Thread.run(Thread.java:761)
07-07 01:16:53.963 30448-31087/org.hisp.dhis.android.trackercapture D/RepoManager: there was an error..HTTP
07-07 01:16:53.963 30448-31087/org.hisp.dhis.android.trackercapture E/RepoManager: {"httpStatus":"Conflict","httpStatusCode":409,"status":"WARNING","message":"One more conflicts encountered, please check import summary.","response":{"responseType":"ImportSummaries","status":"SUCCESS","imported":0,"updated":0,"deleted":0,"ignored":1,"importOptions":{"idSchemes":{},"dryRun":false,"async":false,"importStrategy":"CREATE","mergeMode":"REPLACE","skipExistingCheck":false,"sharing":false,"skipNotifications":false,"datasetAllowsPeriods":false,"strictPeriods":false,"strictCategoryOptionCombos":false,"strictAttributeOptionCombos":false,"strictOrganisationUnits":false,"requireCategoryOptionCombo":false,"requireAttributeOptionCombo":false},"importSummaries":[{"responseType":"ImportSummary","status":"ERROR","description":"TrackedEntityInstance AHXw0cLdpjI already have an active enrollment in program ur1Edk5Oe2n","importCount":{"imported":0,"updated":0,"ignored":1,"deleted":0}}]}}
07-07 01:16:53.965 30448-31087/org.hisp.dhis.android.trackercapture W/CrashlyticsCore: Crashlytics is ignoring a request to log a null exception.
```

- Another possible strategy would to use strategy=CREATE&UPDATE from trackedEntity endpoint as explained in [here](https://docs.dhis2.org/2.25/en/developer/html/dhis2_developer_manual_full.html#d8566e12982)

**Comment by xurxodev** [comment]
@idelcano @ifoche I change status for this issue because sync errors will be solved in #159
